### PR TITLE
Add zsh completion for 'docker daemon --containerd'

### LINKED ISSUE #21539: 1.11.0-rc2: Missing `containerd` flag  in contrib/completion
Containerd integration add a `containerd` flag for docker daemon
<pre><code>--containerd                             Path to containerD socket</code></pre>
it missing in  contrib/completion.
Maybe there are some other flag missing for this release, it needs a check.

ping @albers

### PR DESCRIPTION
**Pull request by sdurrheimer**
Fixes #21539 zsh equivalent

@albers Thx for the ping

Signed-off-by: Steve Durrheimer s.durrheimer@gmail.com

### CODE REVIEW

File: contrib/completion/zsh/_docker [hunk 1]
```diff
@@ -690,6 +690,7 @@ __docker_subcommand() {
                 "($help -b --bridge)"{-b=,--bridge=}"[Attach containers to a network bridge]:bridge:_net_interfaces" \
                 "($help)--bip=[Network bridge IP]:IP address: " \
                 "($help)--cgroup-parent=[Parent cgroup for all containers]:cgroup: " \
+                "($help)--containerd=[Path to containerD socket]:socket:_files -g \"*.sock\"" \
```
Comment (coolljt0725) [inline contrib/completion/zsh/_docker]: `containerD ` should be `containerd`, this typo has been fixed in #21540  :-)

Comment (sdurrheimer) [inline contrib/completion/zsh/_docker]: Fixed !

**Comment by cpuguy83** [comment]
LGTM

**Comment by coolljt0725** [comment]
LGTM

**Pull request merged**